src/main/java/core/mate/academy/service/ExcavatorProducer.java

```diff
@@ -0,0 +1,21 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExcavatorProducer implements MachineProducer<Machine> {
```

Comment:
The `ExcavatorProducer` class should be parameterized with `Excavator` instead of `Machine` to ensure that no implementation not using `Machine` or its subtype can be created.